composer/trainer/checkpoint.py

```diff
@@ -36,7 +36,11 @@ def load_checkpoint(self, state: State):
         self.checkpoint_rng_state = self._get_checkpoint_rng_state(state, self.state_dict["rng"])
 
         if "seed" in self.state_dict:
-            seed_all(self.state_dict["seed"])
+            assert ddp.get_world_size() == len(self.state_dict["seed"]), \
```

Comment:
I would make this into a warning. It's possible that we're restoring on different hardware

Reply:
The issue here is that you won't be able to restore the seeds correctly if the `world_size` doesn't match. If the new `world_size` is smaller but if it is larger then you won't be able to set seeds for the excess processes because there won't be any stored. 

Also, in general if the new `world_size` isn't the same then there can't be any deterministic guarantees when resuming from checkpoint in the case where a seed is set.

Once thing I can do is if the `world_size` doesn't match, then just log a warning and don't restore any seeds - does that sound good @ravi-mosaicml ?

Reply:
Yup, this is fine. I believe the existing behavior is to log a warning and not restore seeds.